platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheGracefulDegradationIntegrationTest.groovy

```diff
@@ -0,0 +1,427 @@
+        configurationCache.assertNoConfigurationCache()
+
+        and:
+        problems.assertResultConsoleSummaryHasNoProblems(result)
+        problems.assertResultHtmlReportHasProblems(result) {
+            totalProblemsCount = 1
+            withProblem("Build file 'included/build.gradle': line 5: invocation of 'Task.project' at execution time is unsupported.")
+            withIncompatibleTask(":included:foo", "Project access.")
+        }
+
+        and:
+        outputContains("Hello from included build :")
+        outputContains("Hello from root build")
+        assertConfigurationCacheDegradation()
+    }
+
+    def "no CC degradation if incompatible task is not presented in the task graph"() {
+        def configurationCache = newConfigurationCacheFixture()
+        buildFile """
+            ${taskWithInjectedDegradationController()}
+            tasks.register("a", DegradingTask) { task ->
+                getDegradationController().requireConfigurationCacheDegradation(task, provider { "Project access" })
+                doLast {
+                    println("Project path is \${project.path}")
+                }
+            }
+
+            tasks.register("b") {
+                doLast {
+                    println "Hello from B"
+                }
+            }
+
+            tasks.all {
+                println "\$it configured"
+            }
+        """
+
+        when:
+        configurationCacheRun "b"
+
+        then:
+        configurationCache.assertStateStored()
+        outputContains("task ':a' configured")
+        outputContains("task ':b' configured")
+        outputContains("Hello from B")
+    }
+
+    def "ignore CC degradation requests at execution time"() {
+        def configurationCache = newConfigurationCacheFixture()
+        buildFile """
+            ${taskWithInjectedDegradationController()}
+            tasks.register("foo", DegradingTask) { task ->
+                def reason = provider { "Misconfiguration!" }
+                doLast {
+                    getDegradationController().requireConfigurationCacheDegradation(task, reason)
+                }
+            }
+        """
+
+        when:
+        configurationCacheRun ":foo", "-d"
+
+        then:
+        configurationCache.assertStateStored()
+
+        and:
+        outputContains("Configuration cache degradation request of task :foo is ignored at execution time")
+    }
+
+    def "tasks instantiated during execution have degradation requests ignored"() {
+        def configurationCache = newConfigurationCacheFixture()
+        buildFile """
+            ${taskWithInjectedDegradationController()}
+            tasks.register("a", DegradingTask) { task ->
+                println("Should be configured")
+                getDegradationController().requireConfigurationCacheDegradation(task, provider { "Project access" })
+            }
+        """
+
+        when:
+        // :tasks instantiates tasks instances at execution time
```

Comment:
❓ I think this is only true when the configuration cache is not enabled. Please check